### ISSUE TITLE
MolarMass instantiated from a partial package

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#924]
**Reported by otter on 8 Dec 2012 22:10 UTC**
Translating model "Modelica.Fluid.Examples.[wiki:PumpingSystem]" in Dymola with option "Advanced.PedanticModelica:=true" results in the error message:

```
Component type specifier MolarMass is from a partial package. That is not correct.
File: Modelica/trunk/Modelica/Media/package.mo, line 4054
Component context: Modelica.Media.Water.simpleWaterConstants.molarMass
Component declared as MolarMass molarMass in Modelica.Media.Interfaces.PartialMedium
```

The error message is correct: simpleWaterConstants is an instance of a record from the partial package partialMedium and this is not allowed.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/924

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 8 Dec 2012 23:30 UTC**
Fixing this needed a number of changes:
 - Type definitions are moved from partialMedium to a non-partial package Types
 - use of record types from partial packages (FluidConstants) removed for water by some restructuring
 - removed the commpletely non-sensical est of TemplateMedium that built on partial types

The result is that Modelica.Media.Examples checks now in Dymola with the pedantic flag set to true.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 8 Dec 2012 23:41 UTC**
DONE in 3071d98cb409abe020ded6c184d9a767dd1422b1
<sub>(unhiding since now fixed and part of the release)</sub>

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#46;se on 10 Dec 2012 16:14 UTC**
Reopening because the change is backwards-incompatible (Modelica.Media.Water.simpleWaterConstants no longer exists).

I recommend the following patch:
```
diff -u -x .svn -r libraries/Modelica 3.2.1/Modelica/Media/Water/package.mo build/lib/omlibrary/Modelica 3.2.1/Media/Water/package.mo
--- libraries/Modelica 3.2.1/Modelica/Media/Water/package.mo	2012-12-10 12:27:02.858268693 +0100
+++ build/lib/omlibrary/Modelica 3.2.1/Media/Water/package.mo	2012-12-10 16:57:47.477790519 +0100
@@ -1,9 +1,9 @@
 within Modelica.Media;
 package Water "Medium models for water"
 
\-
   extends Modelica.Icons.MaterialPropertiesPackage;
 
+  import Modelica.Media.Water.ConstantPropertyLiquidWater.simpleWaterConstants;
 
 package IdealSteam "Water: Steam as ideal gas from NASA source"
 
```

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 14 Dec 2012 17:02 UTC**
Since this is something that would break backward-compatibility it ought to be a blocker.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 14 Dec 2012 17:16 UTC**
Just to be clear here: users have instantiated types from partial packages, which means we have to choose between backwards compatibility and specification compliance in some sense, there  is no way. I would decide in favor of specification compliance, if there is no way to get both. The above compatibility issue is easy to fix, and fixed in my working copy.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 14 Dec 2012 17:23 UTC**
Yes and no, they have used Modelica.Media.Water.simpleWaterConstants, which (being a constant in a non-partial package) should be non-partial. Moving this constant to me seems to be an incompatibility. But one that is rather easy to fix. Give a default renaming import, mark it obsolete.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 25 Mar 2013 21:48 UTC**
Fixed in 1e40f35e204bdd448ef3b2a09e83708f3bbb33d8.

**Comment by modelica-trac-importer** [comment]
**Modified by hubertus on 25 Mar 2013 21:48 UTC**
